inference/core/workflows/core_steps/models/foundation/segment_anything/v1.py

```diff
@@ -0,0 +1,293 @@
+            "block_type": "model",
+        },
+        protected_namespaces=(),
+    )
+    type: Literal["roboflow_core/segment_anything@v1", "SegmentAnything2Model"]
+
+    images: Union[WorkflowImageSelector, StepOutputImageSelector] = ImageInputField
+
+    boxes: StepOutputSelector(
+        kind=[
+            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
+            BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
+            BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
+        ]
+    ) = Field(  # type: ignore
+        description="Boxes (from other model predictions)",
+        examples=["$steps.object_detection_model.predictions"],
+        default=None,
+    )
+
+    sam2_model: Union[
+        WorkflowParameterSelector(kind=[STRING_KIND]),
+        Literal["hiera_large", "hiera_small", "hiera_tiny", "hiera_b_plus"],
+    ] = Field(
+        default="hiera_large",
+        description="Model to be used.  One of hiera_large, hiera_small, hiera_tiny, hiera_b_plus",
+        examples=["hiera_large", "$inputs.openai_model"],
+    )
+
+    @classmethod
+    def accepts_batch_input(cls) -> bool:
+        return True
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(
+                name="predictions",
+                kind=[BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND],
+            ),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.0.0,<2.0.0"
+
+
+class SegmentAnything2BlockV1(WorkflowBlock):
+
+    def __init__(
+        self,
+        model_manager: ModelManager,
+        api_key: Optional[str],
+        step_execution_mode: StepExecutionMode,
+    ):
+        self._model_manager = model_manager
+        self._api_key = api_key
+        self._step_execution_mode = step_execution_mode
+
+    @classmethod
+    def get_init_parameters(cls) -> List[str]:
+        return ["model_manager", "api_key", "step_execution_mode"]
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def run(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+        if self._step_execution_mode is StepExecutionMode.LOCAL:
+            return self.run_locally(images=images, sam2_model=sam2_model, boxes=boxes)
+        elif self._step_execution_mode is StepExecutionMode.REMOTE:
+            raise NotImplementedError(
+                "Remote execution is not supported for Segment Anything."
+            )
+        else:
+            raise ValueError(
+                f"Unknown step execution mode: {self._step_execution_mode}"
+            )
+
+    def run_locally(
+        self,
+        images: Batch[WorkflowImageData],
+        sam2_model: str,
+        boxes: Batch[sv.Detections],
+    ) -> BlockResult:
+
+        predictions = []
+
+        if not boxes:
```

Comment:
`boxes` would be always set, do u check for batch size here? what is the purpose of this construction?

Reply:
currently we have as optional property since SAM2 can predict without prompt...although it's results are much better when prompted, so open to have it be required property instead (also gets around weirdness of how to handle class ids/names when not using object detection boxes as prompts).

Reply:
ok, I got the point and that would almost work as u described, but the field:
```
boxes: StepOutputSelector(
        kind=[
            BATCH_OF_OBJECT_DETECTION_PREDICTION_KIND,
            BATCH_OF_INSTANCE_SEGMENTATION_PREDICTION_KIND,
            BATCH_OF_KEYPOINT_DETECTION_PREDICTION_KIND,
        ]
    ) = Field(  # type: ignore
        description="Boxes (from other model predictions)",
        examples=["$steps.object_detection_model.predictions"],
        default=None,
    )
```
is declared as the value that does not tolerate `None` with `None` as default. Let's fix that, apart from this the approach is fine.

Interestingly, despite my worries about pydantic validation error - pydantic does not complain on manifest init like that:

```python
BlockManifest(
    name="a",
    type="roboflow_core/segment_anything@v1",
    images="$inputs.image",
)
```

probably, defaults are by default excluded from validation based on typing

Reply:
Should we make the type `Optional[StepOutputSelector(...)]`?

It seems to work, and in UI it is categorized under optional properties as is

Reply:
yes, you are right